src/HypreLinearSystem.C

```diff
@@ -1463,18 +1548,124 @@ HypreLinearSystem::hypreIJVectorSetAddToValues()
   auto num_rows_owned = hcApplier->num_rows_owned_;
   auto num_rows_shared = hcApplier->num_rows_shared_;
 
+  HypreDirectSolver* solver = reinterpret_cast<HypreDirectSolver*>(linearSolver_);
+  if (solver->getConfig()->simpleHypreMatrixAssemble()) {
+#if 0
```

Comment:
Is this the code that requires updates to hypre to work?

Reply:
Code wrapped in "#if 0" is code in a branch of Hypre that I have not get merged to Hypre master. I'm keeping it in there to prepare for the time when I do get that into master.

In this PR, simple_hypre_matrix_assemble does nothing. We just use the default Hypre assembly. Once, the Hypre branch is merged, it will allow me to either toggle between the default assembly and the new, fast assembly. 

I'm trying to keep my code from diverging from master too much and not have to rebranch every so often. My intent is definitely to clean this up in the long run while reducing my short term work.